pkg/scaling/resolver/scale_resolvers.go

```diff
@@ -597,6 +606,43 @@ func resolveAuthSecret(ctx context.Context, client client.Client, logger logr.Lo
 	return string(result)
 }
 
+func resolveBoundServiceAccountToken(ctx context.Context, client client.Client, logger logr.Logger, namespace string, bsat *kedav1alpha1.BoundServiceAccountToken, acs *authentication.AuthClientSet) string {
+	serviceAccountName := bsat.ServiceAccountName
+	if serviceAccountName == "" {
+		logger.Error(fmt.Errorf("error trying to get token"), "serviceAccountName is required")
+		return ""
+	}
+	var err error
+
+	serviceAccount := &corev1.ServiceAccount{}
+	err = client.Get(ctx, types.NamespacedName{Name: serviceAccountName, Namespace: namespace}, serviceAccount)
+	if err != nil {
+		logger.Error(err, "error trying to get service account from namespace", "ServiceAccount.Namespace", namespace, "ServiceAccount.Name", serviceAccountName)
+		return ""
+	}
+	return generateToken(ctx, serviceAccountName, namespace, acs)
+}
+
+func generateToken(ctx context.Context, serviceAccountName, namespace string, acs *authentication.AuthClientSet) string {
```

Comment:
Could you please add docs for this function? Give a short overview of what is happening here and/or rename. The name of the function is quite generic so it might be less clear what is it about.